### PR TITLE
cmds/pin: add force option for pin rm

### PR DESCRIPTION
**Pull request by overbool**
Fixes: https://github.com/ipfs/go-ipfs/issues/5716

License: MIT
Signed-off-by: Overbool <overbool.xu@gmail.com>

### CODE REVIEW

**Comment by kevina** [comment]
@overbool could you add a sharness test for this?

File: test/sharness/t0085-pins.sh [hunk 1]
```diff
@@ -56,6 +57,10 @@ test_pins() {
     cat hashes | ipfs pin rm
   '
 
+  test_expect_success "unpin non-existent hashes with force option" '
```
Comment (kevina) [inline test/sharness/t0085-pins.sh]: To make sure things are working right I would do
```
ipfs pin rm --force <invalid-hash> <valid-hash> <invalid-hash>
```

**Comment by overbool** [comment]
@kevina @schomatis Currently, we had implemented the `PinAPI.Rm` in `coreapi/pin.go`, but we haven't used it yet. Should we use `PinAPI.Rm`?

Another question is should we deprecate `corerepo/pinning.go`? Because I think we can implement all function about `pin` in `coreapi/pin.go`.

**Comment by schomatis** [comment]
I'm not sure, @magik6k might know more about that, but those issues seem more suited to be handled in another PR to avoid mixing different solutions.

**Comment by magik6k** [comment]
Sorry for the delay. Yep, we can remove `corerepo/pinning.go` and just use CoreAPI.

**Comment by magik6k** [comment]
I'm guessing we want https://github.com/ipfs/go-ipfs/pull/5843 in first

**Pull request closed without merging**